core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2149,8 +2149,7 @@ private[spark] object Utils
 
   /** Return a heap dump. Used to capture dumps for the web UI */
   def getHeapHistogram(): Array[String] = {
-    // From Java 9+, we can use 'ProcessHandle.current().pid()'
-    val pid = getProcessName().split("@").head
+    val pid = String.valueOf(ProcessHandle.current().pid())
```

Comment:
Yea, this looks more clear.

Reply:
DISCUSS: the new path can result in an Exception being thrown as per documentation.
It is not clear if this can happen for the platforms we support - do we know more about this @dongjoon-hyun ?